### PR TITLE
Allow the agent running in PowerShell to use the PowerShell handler

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
**Description:** fix for [this issue](https://github.com/microsoft/azure-pipelines-tasks/issues/17708). If the agent is running in PowerShell,
the `PSModulePath` environment variable will be removed for tasks executed via the PowerShell handler.

### CODE REVIEW

File: src/Agent.Sdk/Util/ProcessUtil.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace Microsoft.VisualStudio.Services.Agent.Util
+{
+    public static class ProcessUtil
+    {
+        public static Process GetParentProcess(int processId)
+        {
+            try
+            {
+                using var query = new ManagementObjectSearcher($"SELECT * FROM Win32_Process WHERE ProcessId={processId}");
```
[the remark this answers is not in the document]
Comment (KonstantinTyukalov) [reply src/Agent.Sdk/Util/ProcessUtil.cs:17]: if it's extra, please rename this util class into the `WindowsProcessUtil`, or something similar

File: src/Agent.Sdk/Util/ProcessUtil.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+                processList.Add(parentProcess);
+            }
+
+            return processList;
+        }
+
+        public static bool ProcessIsRunningInPowerShell(Process process)
+        {
+            return GetProcessList(process).Exists(process => process.ProcessName.ToLower() == "pwsh" || process.ProcessName.ToLower() == "powershell");
```
Comment (KonstantinTyukalov) [inline src/Agent.Sdk/Util/ProcessUtil.cs:49]: 2 different variables with the same name. Please rename one of them

File: src/Agent.Worker/Handlers/Handler.cs [hunk 3]
```diff
@@ -295,5 +295,14 @@ protected void AddPrependPathToEnvironment()
                 AddEnvironmentVariable(Constants.PathVariable, newPath);
             }
         }
+
+        protected void RemovePSModulePathFromEnvironment()
```
Comment (KonstantinTyukalov) [inline src/Agent.Worker/Handlers/Handler.cs:299]: Should it be inside the `PowerShell3Handler`? Or, why we are not use it in the `PowerShellExeHandler`?

File: src/Agent.Sdk/Util/ProcessUtil.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DenisRumyantsev) [reply src/Agent.Sdk/Util/ProcessUtil.cs:17]: Done, thanks!

File: src/Agent.Sdk/Util/ProcessUtil.cs (hunk 2 above)
Comment (DenisRumyantsev) [reply src/Agent.Sdk/Util/ProcessUtil.cs]: Fixed, thanks!

File: src/Agent.Worker/Handlers/Handler.cs (hunk 3 above)
Comment (DenisRumyantsev) [reply src/Agent.Worker/Handlers/Handler.cs:299]: Added it to the `PowerShellExeHandler` as well, thanks.

File: src/Agent.Sdk/Util/ProcessUtil.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            try
+            {
+                using var query = new ManagementObjectSearcher($"SELECT * FROM Win32_Process WHERE ProcessId={processId}");
+                var process = query.Get().OfType<ManagementObject>().First();
+                var parentProcessId = (int)(uint)process["ParentProcessId"];
+                return Process.GetProcessById(parentProcessId);
+            }
+            catch
+            {
+                return null;
```
Comment (kirill-ivlev) [inline src/Agent.Sdk/Util/ProcessUtil.cs:24]: Maybe it will be better to throw and catch exception instead of returning null?

Comment (DenisRumyantsev) [reply src/Agent.Sdk/Util/ProcessUtil.cs]: Done

**Comment by DenisRumyantsev** [comment]
@KonstantinTyukalov there is no such issue on Linux (tested it).

File: src/Agent.Sdk/Util/ProcessUtil.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (KonstantinTyukalov) [inline src/Agent.Sdk/Util/ProcessUtil.cs:1]: Please rename the file name to match the class name

**Review by KonstantinTyukalov — CHANGES_REQUESTED** [review]

File: src/Agent.Sdk/Util/ProcessUtil.cs (hunk 5 above)
Comment (DenisRumyantsev) [reply src/Agent.Sdk/Util/ProcessUtil.cs:1]: I don't think this is a good idea, as more classes may be added in the future.

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Pull request merged**